### PR TITLE
Added lots of new type hinting/minor fixes

### PR DESCRIPTION
**Pull request by ChadBailey**
* Added some comments
* Fixed up minor issues exposed by type checking
  (i.e. stderr possibly being None)
* Floored spinner progress to prevent erroneous status message

Edit: If anyone is confused I originally opened this under https://github.com/commaai/openpilot/pull/21374 but then accidentally wiped out the commit & pull request by removing the commit from my fork, oops!

The original PR failed to build and initiated an upstream bug report in mypy https://github.com/python/mypy/issues/10695

### CODE REVIEW

File: selfdrive/manager/manager.py [hunk 1]
```diff
@@ -193,6 +215,7 @@ def main():
   if Params().get_bool("DoUninstall"):
     cloudlog.warning("uninstalling")
     HARDWARE.uninstall()
+  return None
```
Comment (adeebshihadeh) [inline selfdrive/manager/manager.py:218]: why was this added?

File: selfdrive/manager/manager.py [hunk 2]
```diff
@@ -177,7 +199,7 @@ def main():
   manager_prepare()
 
   if prepare_only:
-    return
+    return None
```
Comment (adeebshihadeh) [inline selfdrive/manager/manager.py:202]: and this one?

File: selfdrive/manager/manager.py [hunk 3]
```diff
@@ -124,7 +136,7 @@ def manager_thread():
   if os.getenv("NOBOARD") is not None:
     ignore.append("pandad")
   if os.getenv("BLOCK") is not None:
-    ignore += os.getenv("BLOCK").split(",")
+    ignore += os.getenv("BLOCK","").split(",")
```
Comment (adeebshihadeh) [inline selfdrive/manager/manager.py:139]: there's already a check for this. either revert this or remove the if statement

File: common/spinner.py [hunk 4]
```diff
@@ -25,7 +25,10 @@ def update(self, spinner_text: str):
         pass
 
   def update_progress(self, cur: int, total: int):
-    self.update(str(round(100 * cur / total)))
+    # Always floor progress bars because they indicate fully completed chunks
```
Comment (adeebshihadeh) [inline common/spinner.py:28]: no need for the comment here

File: selfdrive/manager/process.py [hunk 5]
```diff
@@ -58,11 +59,11 @@ def join_process(process, timeout):
...
-  daemon = False
-  sigkill = False
-  proc = None
-  enabled = True
+  unkillable: bool = False
+  daemon: bool = False
+  sigkill: bool = False
+  proc: Optional[Process] = None
+  enabled: bool = True
   name = ""
```
Comment (adeebshihadeh) [inline selfdrive/manager/process.py:67]: type hint?

File: selfdrive/timezoned.py [hunk 6]
```diff
@@ -12,7 +13,12 @@
 from selfdrive.swaglog import cloudlog
 
 
-def set_timezone(valid_timezones, timezone):
+def set_timezone(valid_timezones: list, timezone: str):
```
Comment (adeebshihadeh) [inline selfdrive/timezoned.py:16]: annotate return type?

File: selfdrive/manager/manager.py (hunk 1 above)
Comment (ChadBailey) [reply selfdrive/manager/manager.py:218]: mypy doesn't like implied returns, so it will error with either "Missing return statement" or "Return value expected" when adding any return type hints.

File: selfdrive/manager/manager.py (hunk 2 above)
Comment (ChadBailey) [reply selfdrive/manager/manager.py:202]: see prior comments

File: selfdrive/manager/manager.py (hunk 3 above)
Comment (ChadBailey) [reply selfdrive/manager/manager.py:139]: this is another type related oddity... since `os.getenv`'s return value is optional, it would potentially raise an exception in the event of it being None... i couldn't just remove the if statement though because it would add an empty string to the list which is unintended behavior. i'm going to submit a rewritten version of this section to solve all concerns

File: common/spinner.py (hunk 4 above)
Comment (ChadBailey) [reply common/spinner.py:28]: sorry, i've had to have this conversation a surprising amount of times so i suppose i was trying to head it off at the pass

**Comment by ChadBailey** [comment]
i made all the changes based on the feedback... i also renamed watchdog_max_dt to watchdog_max_seconds since dt implies "datetime". from what i was able to find, it looks like it's using nanoseconds downsampled to seconds via / 1e9. let me know if i've gotten that wrong

File: selfdrive/manager/manager.py [hunk 7]
```diff
@@ -123,8 +135,10 @@ def manager_thread():
     ignore += ["manage_athenad", "uploader"]
   if os.getenv("NOBOARD") is not None:
     ignore.append("pandad")
-  if os.getenv("BLOCK") is not None:
-    ignore += os.getenv("BLOCK").split(",")
+  try:
```
Comment (adeebshihadeh) [inline selfdrive/manager/manager.py:138]: revert to the if

**Comment by adeebshihadeh** [comment]
It's dt as in delta time, not datetime.

**Comment by adeebshihadeh** [comment]
Converting to draft until it's ready

**Comment by adeebshihadeh** [comment]
Any update on this? If it's going to sit as a draft for a while, let's close until it's ready.

**Comment by adeebshihadeh** [comment]
Closing due to inactivity. Feel free to reopen after addressing comments.
